stdlib/Sockets/src/addrinfo.jl

```diff
@@ -266,10 +260,15 @@ julia> getipaddrs()
 2-element Array{IPv4,1}:
  ip"10.255.0.183"
  ip"172.17.0.1"
+
+julia> getipaddrs(IPv6)
+2-element Array{IPv6,1}:
+ ip"fe80::9731:35af:e1c5:6e49"
+ ip"fe80::445e:5fff:fe5d:5500"
 ```
 """
-function getipaddrs(include_lo::Bool=false)
-    addresses = IPv4[]
+function getipaddrs(addr_type::Type{T}, include_lo::Bool=false) where T<:IPAddr
+    addresses = IPAddr[]
```

Comment:
```suggestion
    addresses = T[]
```
Since we know the concrete type that's expected.